stdr_robot/src/stdr_robot.cpp

```diff
@@ -152,8 +152,40 @@ namespace stdr_robot
       }
     }
 
-    _motionControllerPtr.reset(
-      new IdealMotionController(_currentPose, _tfBroadcaster, n, getName()));
+    std::string motion_model = result->description.kinematicModel.type;
+    MotionControllerParameters p;
+    p.a_ux_ux = result->description.kinematicModel.a_ux_ux;
+    p.a_ux_uy = result->description.kinematicModel.a_ux_uy;
+    p.a_ux_w = result->description.kinematicModel.a_ux_w;
+
+    p.a_uy_ux = result->description.kinematicModel.a_uy_ux;
+    p.a_uy_uy = result->description.kinematicModel.a_uy_uy;
+    p.a_uy_w = result->description.kinematicModel.a_uy_w;
+
+    p.a_w_ux = result->description.kinematicModel.a_w_ux;
+    p.a_w_uy = result->description.kinematicModel.a_w_uy;
+    p.a_w_w = result->description.kinematicModel.a_w_w;
+
+    p.a_g_ux = result->description.kinematicModel.a_g_ux;
+    p.a_g_uy = result->description.kinematicModel.a_g_uy;
+    p.a_g_w = result->description.kinematicModel.a_g_w;
+
```

Comment:
Since you have already defined a struct (`MotionControllerParameters`), it would be better to have that struct directly initialize from a  `KinematicMsg` or even pass directly that message to the motion controller and not define the struct at all. In that manner, `Robot` would **not** be responsible for filling the parameters.